gnocchi/cli/metricd.py

```diff
@@ -219,6 +224,7 @@ def _get_sacks_to_process(self):
         finally:
             return self._tasks or self.fallback_tasks
 
+    @utils.retry_on_exception.wraps
```

Comment:
That's unrelated to redis, it should probably not be there; it'll retry on ANY exception.